cmd/geth/config.go

```diff
@@ -294,7 +294,105 @@ func makeFullNode(ctx *cli.Context) (*node.Node, ethapi.Backend) {
 		// rather than working correctly until then and freezing at activation until TBC is caught up.
 		// Would rather have node delay full startup to fully sync TBC to required genesis height
 		// than startup without full information and later freeze at hVM Phase 0 activation time.
+
 		for {
+
+			// wait until we have a block header that is above hvm genesis height, then confirm we
+			// have hvm genesis block at the expected height
+			for {
+				select {
+				case <-time.After(1 * time.Second):
+					bh, _, err := vm.TBCFullNode.BlockHeaderBest(ctx.Context)
+					if err != nil {
+						log.Crit(fmt.Sprintf("error getting block header best: %s", err))
+					}
+
+					log.Info(fmt.Sprintf("the current best block height is %d", bh))
+
+					if bh < genesisHeight {
+						continue
+					}
+
+					ghh := genesisHeader.BlockHash()
+					// we're above genesis height, but are we on the best chain with genesis?
+					_, height, err := vm.TBCFullNode.BlockHeaderByHash(ctx.Context, &ghh)
+					if err != nil {
+						log.Crit(fmt.Sprintf("could not get block by hash: %s", err))
+					}
+
+					if height != genesisHeight {
+						log.Crit(fmt.Sprintf("mismatched genesis heights %d != %d", height, genesisHeight))
+					}
+
+					break
+				case <-ctx.Context.Done():
+					return nil, nil
+				}
+				break
+			}
+
+			log.Info("done getting hvm genesis header, waiting until we have all of the blocks")
+
+			// grab the current best block, we will then wait until we have all full blocks up to
+			// this one
+			height, blockHeaderBest, err := vm.TBCFullNode.BlockHeaderBest(ctx.Context)
+			if err != nil {
+				log.Crit(fmt.Sprintf("error getting block header best: %s", err))
+			}
+
+			for {
+				// check if we have the full best block available
+				bhh := blockHeaderBest.BlockHash()
+				available, err := vm.TBCFullNode.FullBlockAvailable(ctx.Context, &bhh)
+				if err != nil {
+					log.Crit(fmt.Sprintf("could not determine if full block available: %s", err))
+				}
+
+				// if it's not available, wait until it is
+				if !available {
+					log.Info(fmt.Sprintf("block not available, will wait: %s:%d", blockHeaderBest.BlockHash().String(), height))
```

Comment:
Should this be `Info`?  We are trying to make logs quieter.

Reply:
yes, will update